url/url.go

```diff
@@ -214,13 +214,15 @@ func (u *URL) fetchParams() {
 		return
 	} else {
 		encodedParams := u.Original[index+1:]
+		if encodedParams == "" {
+			return
+		}
 		u.Params.Decode(encodedParams)
 		u.Original = u.Original[:index]
 	}
 	u.Update()
 }
 
```

Comment:
@dogancanbakir if encodedParams is empty and we return early then `?` will be considered as part of path and it might break MergePath when in unsafe mode. since the goal of `fetchParams` is to remove parameters completely from url for furthur processing . we can instead add new parameter in orderParam like `IncludePrefix` for this special case https://github.com/projectdiscovery/utils/blob/79cdbbcd7cf47d5425f983d67e9777bc3406102a/url/orderedparams.go#L14 and reflect the same in `(OrderedParams).Encode()` while constructing QueryParamString in url and this will be reflected in all downstream dependencies